RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java

```diff
@@ -40,12 +35,11 @@ public void update(ShoppingList shoppingList) throws DataNotFoundException {
         shoppingListDAO.update(shoppingList);
     }
 
-    public void delete(int id) throws DataNotFoundException {
-        ShoppingList shoppingList = shoppingListDAO.getByID(id);
-        if (shoppingList != null){
-            shoppingListDAO.deleteById(id);
+    public void delete(ShoppingList shoppingList) throws DataNotFoundException {
+        if (shoppingList != null) {
+            shoppingListDAO.delete(shoppingList);
         } else {
-            throw new DataNotFoundException(String.format("ShoppingList with id %d was not found", id));
+            throw new DataNotFoundException("ShoppingList was not found");
```

Comment:
But it has some other useful fields, by which it can be identified, for example user id, maybe something else.